### ISSUE TITLE
`existing-relative-links` (`MD057`) - rule is reported multiple times and auto-fix removes closing `)` inside table cell with multiple links

### ISSUE THREAD

**Issue by DetachHead** [rvben/rumdl#727]
### rumdl version

0.2.32

### Installation method

Other

### Bug description

when a table cell contains multiple links where the second link is a relative path that triggers `existing-relative-links`, the rule seems to be reported multiple times, and attempting to fix it via `rumdl fmt` breaks it

### Steps to reproduce

1. create the following `.md` file:
    ```md
    | asdf                      |
    | ------------------------- |
    | [foo](#bar) [](./baz.png) |
    ```
2. run `rumdl fmt`

### Expected behavior

redundant `./` is removed from the second link:
```md
| asdf                      |
| ------------------------- |
| [foo](#bar) [](baz.png) |
```

### Actual behavior

rule is reported multiple times:

```console
$ rumdl fmt asdf.md 
asdf.md:3:18: [MD057] Relative link './baz.png' can be simplified to 'baz.png' [fixed]
asdf.md:3:18: [MD057] Relative link './baz.png' does not exist
asdf.md:3:18: [MD057] Relative link './baz.png' can be simplified to 'baz.png' [fixed]
asdf.md:3:18: [MD057] Relative link './baz.png' does not exist
```

file gets updated to:

```md
| asdf                      |
| ------------------------- |
| [foo](#bar) [](baz.png|
```

i suspect it's because the autofix is being applied twice?

### Configuration

```toml
[global]
enable = ["MD057"]

[MD057]
compact-paths = true
```

### Operating system

Linux

**Comment by rvben** [comment]
Fixed in v0.2.32. The root cause was an unanchored regex search: a bracket whose own destination could not match the URL extraction pattern (fragment-only `(#bar)`, empty `()`) silently borrowed the next link's destination on the same line, which produced the duplicate warnings, and applying the duplicated fix twice corrupted the closing `)` and trailing `|`. Extraction is now anchored to each bracket's own position, fix ranges are deduplicated as defense in depth, and links sharing a line with images are no longer double-validated. Thanks for the precise reproduction, it reproduced byte-for-byte.
